marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowRunnerImpl.java

```diff
@@ -353,25 +353,26 @@ else if (!isJobSuccess.get()) {
                     }
                 }
 
-                JsonNode jobNode;
+                JsonNode jobNode = null;
                 try {
                     jobNode = jobDocManager.getJobs(jobId);
                 }
                 catch (Exception e) {
-                    throw e;
-                }
-
-                ObjectMapper objectMapper = new ObjectMapper();
-                try {
-                    RunFlowResponse jobDoc = objectMapper.treeToValue(jobNode.get("job"), RunFlowResponse.class);
-                    resp.setStartTime(jobDoc.getStartTime());
-                    resp.setEndTime(jobDoc.getEndTime());
-                    resp.setUser(jobDoc.getUser());
-                    resp.setLastAttemptedStep(jobDoc.getLastAttemptedStep());
-                    resp.setLastCompletedStep(jobDoc.getLastCompletedStep());
+                    logger.error(e.getMessage());
                 }
-                catch (Exception e) {
-                    throw new RuntimeException(e);
+                if(jobNode != null) {
+                    ObjectMapper objectMapper = new ObjectMapper();
+                    try {
+                        RunFlowResponse jobDoc = objectMapper.treeToValue(jobNode.get("job"), RunFlowResponse.class);
+                        resp.setStartTime(jobDoc.getStartTime());
+                        resp.setEndTime(jobDoc.getEndTime());
+                        resp.setUser(jobDoc.getUser());
+                        resp.setLastAttemptedStep(jobDoc.getLastAttemptedStep());
+                        resp.setLastCompletedStep(jobDoc.getLastCompletedStep());
+                    }
+                    catch (Exception e) {
+                        logger.error(e.getMessage());
+                    }
                 }
```

Comment:
Nice job with updating start/end time logic on the e-node side! 👍

If you switch the order of the following codes by putting the first snippet before the second snippet, it should be fine.

https://github.com/marklogic/marklogic-data-hub/blob/c7cbd0297c17d5d37ff82abf3157d5959e125cfd/marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowRunnerImpl.java#L356-L376

https://github.com/marklogic/marklogic-data-hub/blob/c7cbd0297c17d5d37ff82abf3157d5959e125cfd/marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowRunnerImpl.java#L338-L354